doc/Architecture-overview/index.md

```diff
@@ -10,156 +10,157 @@ sort_idx: 15
+NoSQL database instance is used to store endpoint-related data that grows linearly as the number of endpoints increases.
 
 NoSQL database nodes can be co-located with Kaa nodes on the same physical or virtual machines, and should be deployed in HA mode for the overall
 high availability of the system.
-Kaa officially supports [Apache Cassandra](http://cassandra.apache.org/) and [MongoDB](https://www.mongodb.com/) as embedded NoSQL database at the moment.
+Kaa officially supports [Apache Cassandra](http://cassandra.apache.org/) and [MongoDB](https://www.mongodb.com/) as the embedded NoSQL database at the moment.
 
-## Internode communications
+### Internode communications
 
 Kaa services use [Apache Thirft](https://thrift.apache.org/) to communicate across processes and nodes.
 Each service obtains metadata about its siblings using [Apache ZooKeeper](https://zookeeper.apache.org/).
-This metadata contain information about Thrift host and port.
+This metadata contains information about the Thrift host and port.
 
-## High availability and scalability
+### High availability and scalability
 
 Kaa cluster scales horizontally and linearly; there is no single point of failure in Kaa cluster architecture.
 Kaa Operations and Bootstrap services are identical and function in active-active HA mode.
-One of the Kaa Control service instances acts as a leader, however, any of the passive ones may be promoted in case of the leader node failure.
+One of the cluster nodes contains an active Control service.
+In case that node fails, a standby Control service in another node is promoted to become active.
 High availability of Kaa Cluster also depends on HA of SQL and NoSQL databases.
 
-## Active load balancing
+### Active load balancing
 
-Kaa SDK choose Bootstrap and Operations service instances pseudo-randomly during session initiation
-(see [Administration guide]({{root_url}}Administration-guide/System-components-overview/) for more details).
+Kaa SDK chooses Bootstrap and Operations service instances pseudo-randomly during session initiation.
+For more details, see [System components overview]({{root_url}}Administration-guide/System-components-overview/).
 
-# Endpoint SDK
+## Endpoint SDK
 
-The Kaa endpoint SDK is a library which provides communication, data marshalling, persistence, and other functions available in Kaa for specific type of an
-endpoint (e.g. [C-based](Using-Kaa-C-endpoint-SDK), [C++-based](Using-Kaa-Cpp-endpoint-SDK), [Java-based](Using-Kaa-Java-endpoint-SDK), [Android-based](),
-[Objective-C-based](Using-Kaa-Objective-C-endpoint-SDK)).
-The client SDK abstracts the communication protocol, data persistence, and other implementation details that may be specific for any concrete solution
-based on Kaa.
+Kaa endpoint SDK is a library that provides communication, data marshalling, persistence, and other functions available in Kaa for specific type of an [endpoint type]({{root_url}}Glossary/#endpoint-type).
```

Comment:
Please replace "... for specific type of an [endpoint type]({{root_url}}Glossary/#endpoint-type)." with "... for specific [type of an endpoint]({{root_url}}Glossary/#endpoint-type)."
Glossary/#endpoint-type not exist in the [KAA-1160 Glossary update started](https://github.com/kaaproject/kaa/pull/872/) PR.